examples/llm/glem.py

```diff
@@ -0,0 +1,440 @@
+"""This example run GLEM model using PyG.
+Original Paper: https://arxiv.org/abs/2210.14709
+“Learning on Large-scale Text-attributed Graphs via Variational Inference“.
+Requirements on top of basic PyG:
+`pip install ogb transformers peft tqdm`.
+GLEM is a data augmentation co-training strategy for LM and GNN, our
+implementation extended original implementation from LM to LLM and opt for LoRA
+from peft.
+
+``note::
+    use addtional trick, please add your external prediction by assigning
+    `ext_pred_path` and combine it into pretraining phase and node features
+"""
+
+import argparse
+import os
+import os.path as osp
+import sys
+import time
+
+import torch
+
+from torch_geometric import seed_everything
+from torch_geometric.data import download_google_url
+from torch_geometric.datasets import TAGDataset
+from torch_geometric.nn.models import GAT, GCN, GLEM, GraphSAGE
+
+# Add the parent directory to sys.path
+parent_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
+sys.path.append(parent_dir)
```

Comment:
Why is this necessary?

Reply:
Removed, originally I used it for unifying data folder for other script.